doc: add info on fota in bt mesh to nRF52/53 ug (#8912)
nrfconnect/sdk-nrf: doc/nrf/ug_bt_mesh_fota.rst

.. _ug_bt_mesh_fota:

Performing FOTA updates in Bluetooth mesh
#########################################

In |NCS|, there is currently no support for Device Firmware Update (DFU) over Bluetooth® mesh when performing firmware over-the-air (FOTA) updates for your Bluetooth mesh devices and applications.
However, support for FOTA updates with point-to-point DFU over Bluetooth Low Energy using the MCUmgr subsystem and the Simple Management Protocol (SMP) is available.

Following the instructions described in :ref:`FOTA over Bluetooth Low Energy<ug_nrf52_developing_ble_fota>`, you can enable the support for and perform FOTA updates using a mobile app.

The :ref:`bluetooth_mesh_light` sample enables support for point-to-point DFU over Bluetooth Low Energy for nRF52 Series development kits using a configuration overlay file :file:`overlay-dfu.conf`.

> FOTA update support is enabled by you simply building the sample with this configuration overlay file and flashing it to your development kit.

Reviewer: The "you" is strange here.

You can then go straight to performing the FOTA update and initiating the DFU process on your device.
Alternatively, see :ref:`FOTA update sample<ug_nrf52_developing_ble_fota>` for details on how to use the :ref:`zephyr:smp_svr_sample` to set up your project to support FOTA updates in |NCS|.

Point-to-point DFU over Bluetooth Low Energy for :ref:`zephyr:thingy53_nrf5340` is supported by default, out-of-the-box.
See :ref:`thingy53_app_update` for more information about updating firmware image on :ref:`zephyr:thingy53_nrf5340`.

.. note::
      When performing FOTA update on a Bluetooth mesh device and the device's composition data is going to change after the update, unprovision the device before downloading the new image.

> If you are using the `nRF Connect Device Manager`_ mobile app to perform FOTA updates, your Bluetooth mesh device might not be visible in the list of available devices.

Author: Moved this around as discussed in a phone call :)

.. _ug_bt_mesh_fota:

Performing FOTA updates in Bluetooth mesh
#########################################

> In |NCS|, there is currently no support for Device Firmware Update (DFU) over Bluetooth® mesh when performing firmware over-the-air (FOTA) updates for your Bluetooth mesh devices and applications.

Reviewer: ```suggestion
In |NCS|, there is currently no support for Device Firmware Update (DFU) over Bluetooth® mesh when performing firmware over-the-air (FOTA) updates for your Bluetooth® mesh devices and applications.
```

> However, support for FOTA updates with point-to-point DFU over Bluetooth Low Energy using the MCUmgr subsystem and the Simple Management Protocol (SMP) is available.

Reviewer: ```suggestion
However, support for FOTA updates with point-to-point DFU over Bluetooth® Low Energy using the MCUmgr subsystem and the Simple Management Protocol (SMP) is available.
```

Following the instructions described in :ref:`FOTA over Bluetooth Low Energy<ug_nrf52_developing_ble_fota>`, you can enable the support for and perform FOTA updates using a mobile app.

> The :ref:`bluetooth_mesh_light` sample enables support for point-to-point DFU over Bluetooth Low Energy for nRF52 Series development kits using a configuration overlay file :file:`overlay-dfu.conf`.

Reviewer: ```suggestion
The :ref:`bluetooth_mesh_light` sample enables support for point-to-point DFU over Bluetooth® Low Energy for nRF52 Series development kits using a configuration overlay file :file:`overlay-dfu.conf`.
```

Author: See previous comment.

> FOTA update support is enabled by you simply building the sample with this configuration overlay file and flashing it to your development kit, and you can go straight to performing the FOTA update and itiating the DFU process on your device.

Reviewer: ```suggestion
FOTA update support is enabled by simply building the sample with this configuration overlay file and flashing it to your development kit. You can then go straight to performing the FOTA update and initiating the DFU process on your device.
```

Alternatively, see :ref:`FOTA update sample<ug_nrf52_developing_ble_fota>` for details on how to use the :ref:`zephyr:smp_svr_sample` to set up your project to support FOTA updates in |NCS|.

> Point-to-point DFU over Bluetooth Low Energy for :ref:`zephyr:thingy53_nrf5340` is supported by default, out-of-the-box.

Reviewer: ```suggestion
Point-to-point DFU over Bluetooth® Low Energy for :ref:`zephyr:thingy53_nrf5340` is supported by default, out-of-the-box.
```

Author: Only necessary on the first occurrence of Bluetooth within a page/file

See :ref:`thingy53_app_update` for more information about updating firmware image on :ref:`zephyr:thingy53_nrf5340`.

.. note::

>       When performing FOTA update on a Bluetooth mesh device and the device's composition data is going to change after the update, unprovision the device before downloading the new image.

Reviewer: ```suggestion
      When performing FOTA update on a Bluetooth® mesh device, the device's composition data is going to change after the update. Unprovision the device before downloading the new image.
```

Author: Will device's composition data always change after the update?

Reviewer: No, not necessarily. But we do not know what happens if you download a new sample with old composition data from a previous sample. I do not think it is necessary if the updated sample has the same composition data layout.

Author: Talked about this with @PavelVPV, and basically this should work... And if not, then this is a bug in his opinion, and it should be fixed.
Let me know if this needs more discussion :)

Disabling the filter
====================

To disable the filter in the `nRF Connect Device Manager`_ mobile app, do the following steps:

1. Tap the :guilabel:`Filter` button at the right top corner of your screen.
#. Deselect :guilabel:`Only devices advertising SMP UUID`.

You should see the device node appear in the list of devices.

Advertising SMP UUID
====================

To make sure that your Bluetooth mesh device advertises the SMP service UUID, you can use the :ref:`bluetooth_mesh_light` sample where this is enabled through configuration overlay file :file:`overlay-dfu.conf`.
Alternatively, do the following:

1. Enable and configure the extended advertising API for your Bluetooth mesh sample.
   To do so, add the following to the :file:`prj.conf` file of your sample::

    CONFIG_BT_EXT_ADV=y
    CONFIG_BT_EXT_ADV_MAX_ADV_SET=2 # 1 set for mesh, 1 set for advertising SMP server
    CONFIG_BT_MAX_CONN=2 # 1 conn for GATT Proxy/PB-GATT, 1 conn for SMP

	.. note::
          As of |NCS| v2.1.2, the extended advertising API is enabled (CONFIG_BT_EXT_ADV=y) by default.

>           You would still need to configure CONFIG_BT_EXT_ADV_MAX_ADV_SET and CONFIG_BT_MAX_CONN as defined above.

Reviewer: Delete the note, as this text is added to 2.2.0 and BT_EXT_ADV is then default.

#. Use the code from the :ref:`zephyr:smp_svr_sample` sample to advertise SMP UUID.

> This will make the device node discoverable by the `nRF Connect Device Manager`_ mobile app with the :guilabel:`Only devices advertising SMP UUID` filter enabled.

Reviewer: ```suggestion
This will make the device discoverable by the `nRF Connect Device Manager`_ mobile app with the :guilabel:`Only devices advertising SMP UUID` filter enabled.
```

> Observe that the device node appears in the list of devices in the mobile app.

Reviewer: ```suggestion
Observe that the device appears in the list of devices in the mobile app.
```
